### PR TITLE
Update devcontainers to align with other RAPIDS repositories

### PR DESCRIPTION
**Pull request by trxcllnt**
Removes the unified and isolated devcontainers in favor of just the single one. The unified devcontainers are [still available](https://github.com/rapidsai/devcontainers/blob/d970ac17a894f5d1047fcfd17cffa6d14fe66ec8/.devcontainer/cuda12.0-conda/devcontainer.json#L23-L31) in the [`rapidsai/devcontainers`](https://github.com/rapidsai/devcontainers) repository.

Adds an optional job to the `pr.yaml` to [build the cuGraph libs in each devcontainer](https://github.com/trxcllnt/cuspatial/blob/fea/devcontainers/.github/workflows/pr.yaml#L106-L111), so the build caches are populated for devs by CI.

Other PRs:

* https://github.com/rapidsai/rmm/pull/1328
* https://github.com/rapidsai/kvikio/pull/273
* https://github.com/rapidsai/cudf/pull/14015
* https://github.com/rapidsai/raft/pull/1791
* https://github.com/rapidsai/cumlprims_mg/pull/149
* https://github.com/rapidsai/cuml/pull/5568
* https://github.com/rapidsai/cugraph-ops/pull/538
* https://github.com/rapidsai/cugraph/pull/3838

### CODE REVIEW

File: dependencies.yaml [hunk 1]
```diff
@@ -238,7 +238,7 @@ dependencies:
     common:
       - output_types: [conda, requirements, pyproject]
         packages:
-          - cython>=0.29,<0.30
+          - cython>=3.0.0
```
Comment (harrism) [inline dependencies.yaml:241]: Can you add clang-tools to this dependencies.yaml?

**Review by harrism — APPROVED** [review]
Thanks for this. Much simpler. Just one request.

File: dependencies.yaml (hunk 1 above)
Comment (trxcllnt) [reply dependencies.yaml:241]: Yep, can do

Comment (trxcllnt) [reply dependencies.yaml:241]: @harrism it [looks like](https://github.com/rapidsai/cuspatial/blob/branch-23.10/.pre-commit-config.yaml#L34) the pre-commit hooks use clang-format 15. Should I file a follow-up to re-format everything with clang-format 16?

Comment (harrism) [reply dependencies.yaml:241]: That seems fine, if it's no trouble.

**Review by AyodeAwe — APPROVED** [review]

**Pull request merged**